src/main/java/com/example/EnumDemo/service/DressService.java

```diff
@@ -0,0 +1,97 @@
+package com.example.EnumDemo.service;
+
+import com.example.EnumDemo.dto.DressDto;
+import com.example.EnumDemo.dto.ShowDressDto;
+import com.example.EnumDemo.entity.DressEntity;
+import com.example.EnumDemo.enumC.Color;
+import com.example.EnumDemo.enumC.Size;
+import com.example.EnumDemo.enumC.Type;
+import com.example.EnumDemo.mapper.DressMapper;
+import com.example.EnumDemo.repository.DressRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class DressService {
+
+    @Autowired
+    private DressMapper dressMapper;
+
+    @Autowired
+    private DressRepository dressRepository;
+
+    public DressDto addDress(@RequestBody DressDto dressDto) {
+        DressEntity dressEntity = new DressEntity();
+        dressEntity = dressMapper.dtoToEntity(dressDto);
+        dressRepository.save(dressEntity);
+        return dressDto;
+    }
+
+    public List<ShowDressDto> filterDresses(List<Color> colors, Color color, Size size, Type type, String brand,
+                                            Double lowerPrice, Double upperPrice, boolean notColor,
+                                            boolean priceBetween, boolean priceLessThanEqual, boolean priceGreaterThanEqual) {
+        if (colors != null && !colors.isEmpty()) {
+            List<DressEntity> dressEntities = dressRepository.findByColorIn(colors);
+            List<ShowDressDto> showDressDtos = dressEntities.stream()
```

Comment:
Instead of iterating on each dress entity in service, you can add a method in mapper that maps list of entities to dtos
This will reduce the repetitive code in this service